packages/@o3r/core/schematics/ng-add/project-setup/index.ts

```diff
@@ -41,7 +41,7 @@ export const prepareProject = (options: NgAddSchematicsSchema) => async (tree: T
     readPackageJson, removePackages, renamedPackagesV7toV8, updateImports, isMultipackagesContext, getO3rPeerDeps
   } = await import('@o3r/schematics');
   const installOtterLinter = await shouldOtterLinterBeInstalled(context);
-  const workspaceProject = tree.exists('angular.json') ? getProjectFromTree(tree, options.projectName) : undefined;
+  const workspaceProject = (tree.exists('angular.json') && options.projectName) ? getProjectFromTree(tree, options.projectName) : undefined;
```

Comment:
the function `getProjectFromTree` should be deprecated, the implementation that defaults on the first of the list should be delegated to the schematic.
I would suggest to do a `options.projectName && getWorkspaceConfig(tree)?.projects?.[options.projectName] || undefined`

Reply:
IMO this should be a dedicated task, as **getProjectFromTree** is extensively used across the schematic scripts of updates and in generators.